src/sentry/static/sentry/app/__mocks__/api.jsx

```diff
@@ -32,14 +32,19 @@ class Client {
         headers: response.headers || {},
       },
       mock,
+      predicate,
     ]);
 
     return mock;
   }
 
   static findMockResponse(url, options) {
-    return Client.mockResponses.find(([response]) => {
-      return url === response.url && (options.method || 'GET') === response.method;
+    return Client.mockResponses.find(([response, mock, predicate]) => {
```

Comment:
Would having predicate be an option be simpler? Callers wouldn't need to name the function to have readable code, nor would they need to handle multiple arguments.

Reply:
@markstory I'm unsure what you meant, do you mean something like this? https://github.com/getsentry/sentry/pull/13917/commits/64980cc2051622f16f45e17a7fd625684af3d6f3

Reply:
Exactly that :+1:

Reply:
gotcha